src/main.cpp

```diff
@@ -1726,56 +1714,15 @@ int64_t GetConstantBlockReward(const CBlockIndex* index)
 }
 
 int64_t GetProofOfStakeReward(
-    uint64_t nCoinAge,
-    int64_t nFees,
-    const NN::MiningId mining_id,
-    int64_t nTime,
-    const CBlockIndex* pindexLast,
-    int64_t& out_research_subsidy,
-    int64_t& out_block_subsidy)
+    const uint64_t nCoinAge,
+    const int64_t nTime,
+    const CBlockIndex* const pindexLast)
 {
-    // Research Age Subsidy - PROD
-    out_research_subsidy = 0;
-    out_block_subsidy = 0;
-
-    // Tally doesn't calculate research age averages until block version 9:
-    //
-    if (pindexLast->nVersion >= 9) {
-        if (const NN::CpidOption cpid = mining_id.TryCpid()) {
-            const NN::ResearchAccount& account = NN::Tally::GetAccount(*cpid);
-            const NN::AccrualComputer calc = NN::Tally::GetComputer(*cpid, nTime, pindexLast);
-
-            out_research_subsidy = calc->Accrual(account);
-        }
-    }
-
-    /* Constant Block Reward */
-    if (pindexLast->nVersion>=10)
-        out_block_subsidy = GetConstantBlockReward(pindexLast);
-    else
-        out_block_subsidy = nCoinAge * GetCoinYearReward(nTime) * 33 / (365 * 33 + 8);
-
-    int64_t nSubsidy = out_block_subsidy + out_research_subsidy;
-
-    if (fDebug10 || GetBoolArg("-printcreation"))
-    {
-        LogPrintf("GetProofOfStakeReward(): create=%s nCoinAge=%" PRIu64 " research=%s",
-            FormatMoney(nSubsidy), nCoinAge, FormatMoney(out_research_subsidy));
-    }
-
-    int64_t nTotalSubsidy = nSubsidy + nFees;
-    // This rule does not apply in v11
-    if (out_research_subsidy > 1 && pindexLast->nVersion <= 10)
-    {
-        std::string sTotalSubsidy = RoundToString(CoinToDouble(nTotalSubsidy)+.00000123,8);
-        if (sTotalSubsidy.length() > 7)
-        {
-            sTotalSubsidy = sTotalSubsidy.substr(0,sTotalSubsidy.length()-4) + "0124";
-            nTotalSubsidy = RoundFromString(sTotalSubsidy,8)*COIN;
-        }
```

Comment:
Ah, that was before my time, thanks. If I'm reading #1390 correctly, this change shouldn't impact old clients because they will continue to generate blocks with the indicator, and the old nodes only see their own generated blocks. Was there somewhere else it's used that I don't know of?

It won't hurt to put it back, though. That's probably the safest option.

Reply:
You know, I think you are right. The nodes (IsMine) only see their own RR transactions, and therefore if an old node is running, it will properly see RR transactions generated by itself because they have the indicator.

Reply:
I think we can leave it as it is.